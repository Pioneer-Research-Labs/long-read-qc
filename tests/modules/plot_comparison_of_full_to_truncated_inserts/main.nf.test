nextflow_process {

    name "Test Process plot_comparison_of_full_to_truncated_inserts"
    script "modules/plot_comparison_of_full_to_truncated_inserts/main.nf"
    process "plot_comparison_of_full_to_truncated_inserts"

    test("Should generate a file containing 4 plots") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/inserts.fasta'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/inserts_from_truncated_flanks.fasta'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/untrimmed_from_truncated_flanks.fastq'),
                file("${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz"))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/Gateway_EmptyBarcodedLib1_20241217/truncated_vs_intact_flanks_comparison_plot.png").match()
        }

    }

}
