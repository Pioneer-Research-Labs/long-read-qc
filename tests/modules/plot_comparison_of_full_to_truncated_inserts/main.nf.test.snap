{
    "Should generate a file containing 4 plots": {
        "content": [
            "truncated_vs_intact_flanks_comparison_plot.png:md5,6dd717033bcc92525f3b50707e846317"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-30T16:18:08.982602"
    }
}